sys/suit/v4/handlers.c

```diff
@@ -0,0 +1,548 @@
+        return res;
+    }
+
+    manifest->state |= SUIT_MANIFEST_HAVE_IMAGE;
+
+    return SUIT_OK;
+}
+
+static int _version_handler(suit_v4_manifest_t *manifest, int key,
+                            CborValue *it)
+{
+    (void)manifest;
+    (void)key;
+    /* Validate manifest version */
+    int version = -1;
+    if (cbor_value_is_integer(it) &&
+        (cbor_value_get_int(it, &version) == CborNoError)) {
+        if (version == SUIT_VERSION) {
+            manifest->validated |= SUIT_VALIDATED_VERSION;
+            LOG_INFO("suit: validated manifest version\n)");
+            return 0;
+        }
+        else {
+            return -1;
+        }
+    }
+    return -1;
+}
+
+static int _seq_no_handler(suit_v4_manifest_t *manifest, int key, CborValue *it)
+{
+    (void)manifest;
+    (void)key;
+    (void)it;
+
+    int64_t seq_nr;
+    if (cbor_value_is_unsigned_integer(it) &&
+        (cbor_value_get_int64_checked(it, &seq_nr) == CborNoError)) {
+
+        const riotboot_hdr_t *hdr = riotboot_slot_get_hdr(riotboot_slot_current());
+        if (seq_nr <= (int64_t)hdr->version) {
+            LOG_INFO("%"PRIu64" <= %"PRIu32"\n", seq_nr, hdr->version);
+            LOG_INFO("seq_nr <= running image\n)");
+            return -1;
+        }
+
+        hdr = riotboot_slot_get_hdr(riotboot_slot_other());
+        if (riotboot_hdr_validate(hdr) == 0) {
+            if (seq_nr <= (int64_t)hdr->version) {
+                LOG_INFO("%"PRIu64" <= %"PRIu32"\n", seq_nr, hdr->version);
+                LOG_INFO("seq_nr <= other image\n)");
+                return -1;
+            }
+        }
+
+        LOG_INFO("suit: validated sequence number\n)");
+        manifest->validated |= SUIT_VALIDATED_SEQ_NR;
+        return 0;
+    }
+    LOG_INFO("Unable to get sequence number\n");
+    return -1;
+}
+
+static int _dependencies_handler(suit_v4_manifest_t *manifest, int key,
+                                 CborValue *it)
+{
+    (void)manifest;
+    (void)key;
+    (void)it;
+    /* No dependency support */
+    return 0;
+}
+
+static int _component_handler(suit_v4_manifest_t *manifest, int key,
+                              CborValue *it)
+{
+    (void)manifest;
+    (void)key;
+
+    CborValue arr;
+
+    LOG_DEBUG("storing components\n)");
+    if (!cbor_value_is_array(it)) {
+        LOG_DEBUG("components field not an array\n");
+        return -1;
+    }
+    cbor_value_enter_container(it, &arr);
+
+    unsigned n = 0;
+    while (!cbor_value_at_end(&arr)) {
+        CborValue map, key, value;
+        if (n < SUIT_V4_COMPONENT_MAX) {
+            manifest->components_len += 1;
+        }
+        else {
+            LOG_DEBUG("too many components\n)");
+            return SUIT_ERR_INVALID_MANIFEST;
+        }
+
+        suit_cbor_map_iterate_init(&arr, &map);
+
+        suit_v4_component_t *current = &manifest->components[n];
+
+        while (suit_cbor_map_iterate(&map, &key, &value)) {
+            // handle key, value
```

Comment:
Should be C-style comment. There's another one below.

Reply:
yup